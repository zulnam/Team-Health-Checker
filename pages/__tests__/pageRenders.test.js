import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import About from '../about';
import CreateRoom from '../createRoom';
import Index from '../index';
import Room from '../room';

describe('All pages render correctly', () => {
  it('Renders root page', () => {
    const { container } = render(<Index />);
    expect(container).toMatchSnapshot();
  });

  it('Renders about page', () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot();
  });

  it('Renders the create room page', () => {
    const { container } = render(<CreateRoom />);
    expect(container).toMatchSnapshot();
  });

  it('Renders the room session page', () => {
    const options = {
      router: {
        query: {
          teamName: 'Test Team',
          roomName: 'Test Room'
        }
      }
    };

    const wrapper = shallow(<Room />, options);
    expect(wrapper).toMatchSnapshot();
  });
});
