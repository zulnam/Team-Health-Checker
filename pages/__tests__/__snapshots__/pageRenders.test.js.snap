// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All pages render correctly Renders about page 1`] = `
<div>
  <div
    class="css-z0uj6b-MainContainer e1d4bcbp0"
  >
    <div
      class="css-jog5gk-Container e1gg83400"
    >
      <a
        class="css-16huv58-Title e1gg83401"
      >
        TEAM HEALTH CHECKER
      </a>
      <div
        class="css-1gv4faj-NavBar e1gg83402"
      >
        <ul>
          <li>
            <a
              href="/createRoom"
            >
              Create Room
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/reports"
            >
              Reports
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div>
      <p>
        Welcome to the Team Health Checker tool. Please see the table of contents below for further information.
      </p>
      <ol>
        <li>
          Initiating a room
        </li>
        <li>
          Using a room as an admin
        </li>
        <li>
          Using a room as a user
        </li>
        <li>
          Viewing reports
        </li>
        <li>
          Exporting reports
        </li>
      </ol>
    </div>
  </div>
</div>
`;

exports[`All pages render correctly Renders root page 1`] = `
<div>
  <div
    class="css-z0uj6b-MainContainer e1d4bcbp0"
  >
    <div
      class="css-jog5gk-Container e1gg83400"
    >
      <a
        class="css-16huv58-Title e1gg83401"
      >
        TEAM HEALTH CHECKER
      </a>
      <div
        class="css-1gv4faj-NavBar e1gg83402"
      >
        <ul>
          <li>
            <a
              href="/createRoom"
            >
              Create Room
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/reports"
            >
              Reports
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`All pages render correctly Renders the create room page 1`] = `
<div>
  <div
    class="css-z0uj6b-MainContainer e1d4bcbp0"
  >
    <div
      class="css-jog5gk-Container e1gg83400"
    >
      <a
        class="css-16huv58-Title e1gg83401"
      >
        TEAM HEALTH CHECKER
      </a>
      <div
        class="css-1gv4faj-NavBar e1gg83402"
      >
        <ul>
          <li>
            <a
              href="/createRoom"
            >
              Create Room
            </a>
          </li>
          <li>
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li>
            <a
              href="/reports"
            >
              Reports
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div>
      <form
        class="css-fnd3so-FormContainer eiytlan0"
      >
        <article
          class="css-3g7ek1-FormField eiytlan1"
        >
          <label
            for="roomName"
          >
            Room Name: 
          </label>
          <input
            data-testid="roomName"
            name="roomName"
            required=""
            type="text"
            value=""
          />
        </article>
        <article
          class="css-3g7ek1-FormField eiytlan1"
        >
          <label
            for="teamName"
          >
            Tribe: 
          </label>
          <select
            data-testid="teamName"
            name="teamName"
          >
            <option
              value=""
            >
              Select a Team
            </option>
            <option
              value="Browse"
            >
              Browse
            </option>
            <option
              value="Trade"
            >
              Trade
            </option>
            <option
              value="Buy"
            >
              Buy
            </option>
          </select>
        </article>
        <a
          href="/room?roomName=&teamName="
        >
          Create Room
        </a>
      </form>
    </div>
  </div>
</div>
`;

exports[`All pages render correctly Renders the room session page 1`] = `ShallowWrapper {}`;
